model/canvas.cc

```diff
@@ -145,9 +145,25 @@ namespace minsky
       }
     if (fromPort.get())
       {
-          if (auto to=closestInPort(x,y))
+          if (auto to=closestInPort(x,y)) {
             model->addWire(static_cast<shared_ptr<Port>&>(fromPort),to);
-        fromPort.reset();
+            fromPort.reset();
+            auto& toItem=to->item();
+            if (auto ravel=dynamic_cast<Ravel*>(&toItem))
+            {
+              auto state = ravel->getState();
+              if(state.empty()) {
+                minsky().reset();
```

Comment:
This is OK for now, but I had another thought overnight. In addon.cc, there is a separate thread that runs when the backend has been idle for a little  - around 100ms I think. Why not check the state of the reset_needed flag, and call reset at that point. That way, if reset takes a little while (which will be the case for some extensive tensor operations), it will not interfere with model responsiveness when doing multiple mousing operations on the screen.

I'm a little worried about the call to reset on line 144 for similar reasons too.